lib/webauthn/public_key_credential_with_attestation.rb

```diff
@@ -5,11 +5,15 @@
 
 module WebAuthn
   class PublicKeyCredentialWithAttestation < PublicKeyCredential
+    class InvalidChallengeError < Error; end
+
     def self.response_class
       WebAuthn::AuthenticatorAttestationResponse
     end
 
     def verify(challenge, user_verification: nil)
+      challenge.is_a?(String) || raise(InvalidChallengeError, "challenge must be a String. input challenge class: #{challenge.class}")
```

Comment:
What do you think about using the [already existent `ChallengeVerificationError`](https://github.com/cedarcode/webauthn-ruby/blob/23f3be4/lib/webauthn/authenticator_response.rb#L13)?

Reply:
@santiagorodriguez96 

Indeed, I think it is reasonable to handle `WebAuthn::ChallengeVerificationError` because the meaning of `ChallengeVerificationError` and the defined level are directly under `WebAuthn`.

- The name and responsibility of `ChallengeVerificationError` is broad. The `WebAuthn::PublicKeyCredentialWithAttestation::InvalidChallengeError` added this time has a limited `class` definition and naming, so it helps detailed error analysis.
- The usage of `ChallengeVerificationError` is unified in that the `class` is dynamically selected by the `verify_item` method. This time, explicitly specifying a different `class` for a different purpose may cause confusion for users.

After that, I recommend defining a new `class` like the current implementation.
And, either choice makes sense, so I respect your opinion.

Reply:
Fair enough, there are other cases where we are raising a custom error, for example in https://github.com/cedarcode/webauthn-ruby/blob/23f3be4/lib/webauthn/attestation_statement/base.rb#L163

Reply:
thank you. If that's the case, I think I'd like to use a custom error this time as well for the reasons mentioned above.

Reply:
thank you. In that case, for the reasons mentioned above, I thought it would be better to add a custom error, which is the current modification content.
What do you think after sharing my opinion?

Reply:
Some part of me thinks that receiving a `nil` challenge in some sense it is a challenge verification error, thus my initial idea of just re-using the `ChallengeVerificationError` and, as you are correctly doing, use the message to be specific about what was the underlying issue with the `challenge`.

However, I do see how using `VerificationError` classes could cause confusion given that, as you correctly pointed out, `VerificationError`s are dynamically raised depending on which verification was the one that failed.

Furthermore, we are already using the pattern of raising an error if there's a validation error, so I'm okay with the approach.